### ISSUE TITLE
[Win32] FileDialog ignores set filter path (directory)

### ISSUE THREAD

**Issue by tmssngr** [eclipse-platform/eclipse.platform.swt#529]
**Describe the bug**
When opening a FileDialog, the set filter path (directory) is ignored.

**To Reproduce**
Please launch this snippet and click the **Open File Dialog** button. It should take the directory from the text field (which defaults to the current directory).
```java
import java.nio.file.Path;
import java.nio.file.Paths;

import org.eclipse.swt.SWT;
import org.eclipse.swt.layout.GridData;
import org.eclipse.swt.layout.GridLayout;
import org.eclipse.swt.widgets.Button;
import org.eclipse.swt.widgets.Display;
import org.eclipse.swt.widgets.FileDialog;
import org.eclipse.swt.widgets.Shell;
import org.eclipse.swt.widgets.Text;

public class FileDialogTest {

	public static void main(String[] args) {
		final Display display = new Display();

		final Shell shell = new Shell(display);
		shell.setLayout(new GridLayout(2, false));

		final Text directoryText = new Text(shell, SWT.BORDER | SWT.SINGLE);
		directoryText.setText(System.getProperty("user.dir"));
		directoryText.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));

		final Button button = new Button(shell, SWT.PUSH);
		button.setText("Open File Dialog...");
		button.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, false));

		button.addListener(SWT.Selection, event -> {
			final FileDialog dialog = new FileDialog(shell, SWT.OPEN | SWT.PRIMARY_MODAL);

			final String path = directoryText.getText();
			dialog.setFilterPath(path);
			dialog.setFileName("my file.txt");

			final String selectedPathString = dialog.open();
			if (selectedPathString == null) {
				return;
			}

			final Path selectedPath = Paths.get(selectedPathString);
			directoryText.setText(selectedPath.getParent().toString());
		});

		shell.setSize(400, 200);
		shell.open();

		while (!shell.isDisposed()) {
			if (!display.readAndDispatch()) {
				display.sleep();
			}
		}

		display.dispose();
	}
}
```
But for me it shows a completely other unrelated directory.

**Expected behavior**
The file dialog should initially show the specified directory. It looks like Windows remembers the last used file chooser path and shows that instead, even after restarting the application.

**Screenshots**
![screenshot](https://user-images.githubusercontent.com/6029561/212750261-b92a6fcf-7f47-4566-9ac5-5ec7d27042ab.png)

**Environment:**
1. Select the platform(s) on which the behavior is seen:
- - [ ] All OS
- - [x] Windows
- - [ ] Linux
- - [ ] macOS 

2. Additional OS info (e.g. OS version, Linux Desktop, etc)
I can reproduce the problem on Windows 10 and Windows 11.

**Workaround (or) Additional context**
I'm not sure whether this is a bug in SWT (in how the Windows API is used) or a bug in Windows.

**Comment by Phillipus** [comment]
Perhaps because of [this](https://github.com/eclipse-platform/eclipse.platform.swt/commit/e84f26b436d4c04e11d0797e013c63318f2638c1)?

**Comment by tmssngr** [comment]
Thanks. Changing `SetDefaultFolder` to `SetFolder` fixes this bug.

**Comment by Phillipus** [comment]
So with the "fix" in https://github.com/eclipse-platform/eclipse.platform.swt/commit/e84f26b436d4c04e11d0797e013c63318f2638c1 `setFilterPath` is useless?

**Comment by tmssngr** [comment]
Yes.

**Comment by tmssngr** [comment]
> > Thanks. Changing `SetDefaultFolder` to `SetFolder` fixes this bug.
> 
> I can't follow, _where_ this should be set?

Please look at https://github.com/syntevo/eclipse.platform.swt/commit/e2301ff5af27ef08b1c5fbc0170b2e51a6c021c5

**Comment by iloveeclipse** [comment]
> > > Thanks. Changing `SetDefaultFolder` to `SetFolder` fixes this bug.
> > 
> > 
> > I can't follow, _where_ this should be set?
> 
> Please look at [syntevo@e2301ff](https://github.com/syntevo/eclipse.platform.swt/commit/e2301ff5af27ef08b1c5fbc0170b2e51a6c021c5)

I still can't follow. The links points to custom fork. This is the bug tracker of the official SWT project, not of a fork.

The question I have is: do we need to revert something in the **original** SWT code or is this issue about some **custom** SWT fork code and can be closed?

**Comment by tmssngr** [comment]
This bug is a regression of commit e84f26b4. I've fixed it in our own fork of the SWT repository. Hence, for me the problem is solved, but for all the users of the original SWT code the regression remains.

**Comment by iloveeclipse** [comment]
> I've fixed it in our own fork of the SWT repository.

... and what prevents you to create a PR with that change for the official SWT code?

**Comment by lshanmug** [comment]
Please see https://github.com/eclipse-platform/eclipse.platform.swt/pull/466 for additional discussion and alternate PR.

**Comment by niraj-modi** [comment]
> This issue is fixed via the pr #556 => #556
> 
> The same fix is now available in RC1.
> 
> So you may close this issue now.

Thanks @deepika-u for the fix, closing now.
